core/src/main/java/nordpol/Apdu.java

```diff
@@ -97,4 +101,103 @@ public static String encodeHex(byte b) {
         return Arrays.copyOfRange(response, 0, response.length - 2);
     }
 
+    /**
+     * Matches the first APDUs last characters (of the length of the second APDU) against the second APDU
+     * @param receivedApdu The response APDU as byte[] to be compared
+     * @param expectedApdu The expected APDU as byte[] to match against
+     * @return true if the APDUs match, false if not
+     */
+    public static boolean hasStatus(byte[] receivedApdu, byte[] expectedApdu) throws IOException {
+        return hasStatus(encodeHex(receivedApdu), encodeHex(expectedApdu));
+    }
+
+    /**
+     * Matches the first APDUs last characters (of the length of the second APDU) against the second APDU
+     * @param receivedApdu The response APDU as byte[] to be compared
+     * @param expectedApdu The expected APDU as String to match against
+     * @return true if the APDUs match, false if not
+     */
+    public static boolean hasStatus(byte[] receivedApdu, String expectedApdu) throws IOException {
+        return hasStatus(encodeHex(receivedApdu), expectedApdu);
+    }
+
+    /**
+     * Matches the first APDUs last characters (of the length of the second APDU) against the second APDU
+     * @param receivedApdu The response APDU as String to be compared
+     * @param expectedApdu The expected APDU as String to match against
+     * @return true if the APDUs match, false if not
+     */
+    public static boolean hasStatus(String receivedApdu, String expectedApdu) throws IOException {
+        String concatReceivedApdu = shortenString(receivedApdu, expectedApdu.length());
+        return concatReceivedApdu.equalsIgnoreCase(expectedApdu);
+    }
+
+    /**
+     * Transceives the byte[] command to the IsoCard. Automatically buffers the rest of the message if it
```

Comment:
@Franzaine I think the comment is a bit misleading. (sorry for not commenting on this earlier) it Should be something like "Automatically queries for more response data if the status code indicates that more data is available"